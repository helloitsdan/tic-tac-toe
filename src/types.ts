export enum BoardPosition {
  TOP_LEFT,
  TOP_CENTRE,
  TOP_RIGHT,
  MIDDLE_LEFT,
  MIDDLE_CENTRE,
  MIDDLE_RIGHT,
  BOTTOM_LEFT,
  BOTTOM_CENTRE,
  BOTTOM_RIGHT,
}

export enum PlayerPiece {
  PLAYER_ONE = 'PLAYER_ONE',
  PLAYER_TWO = 'PLAYER_TWO',
}

export enum MatchName {
  TOP_ROW = 'TOP_ROW',
  MIDDLE_ROW = 'MIDDLE_ROW',
  BOTTOM_ROW = 'BOTTOM_ROW',
  LEFT_COLUMN = 'LEFT_COLUMN',
  CENTRE_COLUMN = 'CENTRE_COLUMN',
  RIGHT_COLUMN = 'RIGHT_COLUMN',
  DIAGONAL_LEFT_TO_RIGHT = 'DIAGONAL_LEFT_TO_RIGHT',
  DIAGONAL_RIGHT_TO_LEFT = 'DIAGONAL_RIGHT_TO_LEFT',
}

export interface PlayerType {
  piece: PlayerPiece
  name: string
}

export type MatchTuple = [MatchName, BoardPosition[]]

export interface PlayerMatchType {
  match: MatchTuple
  piece: PlayerPiece
}

export type GameMovesType = Record<PlayerPiece, BoardPosition[]>

export interface GameStateType {
  players: PlayerType[]
  moves: GameMovesType
  winner: PlayerMatchType
}
