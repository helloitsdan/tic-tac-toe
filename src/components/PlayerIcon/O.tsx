import React, { FunctionComponent } from 'react'
import { ReactComponent as OIcon } from '../../assets/O.svg'

const O: FunctionComponent = () => (
  <OIcon title="O" className="c-player-piece" />
)

export default O
