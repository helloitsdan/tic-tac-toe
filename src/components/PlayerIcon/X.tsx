import React, { FunctionComponent } from 'react'
import { ReactComponent as XIcon } from '../../assets/X.svg'

const X: FunctionComponent = () => (
  <XIcon title="X" className="c-player-piece" />
)

export default X
