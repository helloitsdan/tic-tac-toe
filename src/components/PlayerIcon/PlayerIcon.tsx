import React, { FunctionComponent } from 'react'
import { PlayerId } from '../../types'
import X from './X'
import O from './O'

interface PlayerIconProps {
  player?: PlayerId
  className?: string
}

const PlayerIcon: FunctionComponent<PlayerIconProps> = ({ player }) => {
  switch (player) {
    case PlayerId.PLAYER_ONE:
      return <X />
    case PlayerId.PLAYER_TWO:
      return <O />
    default:
      return <span>&nbsp;</span>
  }
}

export default PlayerIcon
