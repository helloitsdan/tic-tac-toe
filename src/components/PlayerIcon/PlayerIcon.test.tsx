import React from 'react'
import { render } from '@testing-library/react'

import PlayerIcon from './PlayerIcon'
import { PlayerId } from '../../types'

describe('PlayerIcon', () => {
  it('renders', () => {
    const { asFragment } = render(<PlayerIcon />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders an icon for player one', () => {
    const { getByTitle } = render(<PlayerIcon player={PlayerId.PLAYER_ONE} />)
    expect(getByTitle('X')).not.toBeNull()
  })

  it('renders an icon for player two', () => {
    const { getByTitle } = render(<PlayerIcon player={PlayerId.PLAYER_TWO} />)
    expect(getByTitle('O')).not.toBeNull()
  })
})
