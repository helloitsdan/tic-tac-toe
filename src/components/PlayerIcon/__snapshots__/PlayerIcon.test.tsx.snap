// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerIcon renders 1`] = `
<DocumentFragment>
  <span>
    Â 
  </span>
</DocumentFragment>
`;
